test/controllers/resident/properties/GainControllerSpec/ImprovementsActionSpec.scala

```diff
@@ -194,5 +196,17 @@ class ImprovementsActionSpec extends CommonPlaySpec with WithCommonFakeApplicati
         doc.title() shouldEqual s"Error: ${messages.title}"
       }
     }
+
+    "a NoSuchElementException is thrown" should {
+      "return an ApplicationException" in {
+        when(mockSessionCacheService.fetchAndGetFormData[OwnerBeforeLegislationStartModel](ArgumentMatchers.eq(keystoreKeys.ownerBeforeLegislationStart))
+          (ArgumentMatchers.any(), ArgumentMatchers.any()))
+          .thenReturn(Future.successful(None))
+        val result = intercept[ApplicationException](await(testingGainController.submitImprovements(fakeRequestWithSession)))
+
+        result.message shouldBe "cgt-calculator-resident-properties-frontendNone.get"
```

Comment:
Why do we want None.get as a message.

Reply:
it's because that is what is returned by the SUT, it's an existing unchanged behaviour

Reply:
So I'm not sure whose or which  'Option ' is referred here that is handled by None.get